store/cache/store.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"time"
+
+	xredis "github.com/go-redis/redis/v8"
+	"github.com/tigrisdata/tigris/internal"
+	"github.com/tigrisdata/tigris/server/config"
+)
+
+const (
+	// StreamFromCurrentPos is for creating a consumer group that sets the position as current.
+	StreamFromCurrentPos = "$"
+
+	// StreamFromCurrentPosRead is to let stream know that reads needs to happen from current.
+	StreamFromCurrentPosRead = ">"
+)
+
+type SetOptions struct {
+	// NX is SetIfNotExists i.e. only set the key if it does not already exist.
+	NX bool
+	// XX is SetIfExists i.e. only set the key if it already exists.
+	XX bool
+	// EX sets the Expiry time of the key in seconds
+	EX time.Duration
+}
+
+type GetOptions struct {
+	// Expiry set the expiration time of the key
+	Expiry    time.Duration
+	GetDelete bool
+}
+
+type Stream interface {
```

Comment:
should this be in the stream.go?

Reply:
Either way is fine, Cache interface is creating a Stream so kept it here so it is easy to know what interface is returned when we create a stream.